alacritty/src/display/mod.rs

```diff
@@ -552,9 +558,59 @@ impl Display {
         }
     }
 
-    pub fn make_current(&self) {
-        if !self.context.get().is_current() {
-            self.context.make_current(&self.surface).expect("failed to make context current")
+    pub fn make_current(&mut self, recover: bool) {
+        let is_current = self.context.get().is_current();
+        if is_current && !recover {
+            return;
+        }
+
+        let mut was_context_reset = if is_current {
+            false
+        } else {
+            match self.context.make_current(&self.surface) {
+                Err(err) if err.error_kind() == ErrorKind::ContextLost => {
+                    info!("Context lost for window {:?}", self.window.id());
+                    true
+                },
+                _ => false,
+            }
+        };
+
+        was_context_reset |= self.renderer.was_context_reset();
```

Comment:
Is there a realistic benefit to this not just being inside the `is_current` branch? Sure it might help if the context resets right after the `make_current` call, but that could also happen right after this call, right? So in the end this should only really be necessary if `is_current` is true?

Reply:
I feel like it could lie/desync/not propagate, it's opengl at the end of the day.

Reply:
But isn't the same true for this function too? How likely are the two to be different?

Reply:
The context is done by WSI, but the OpenGL is by driver itself, though, eventually the `is_current` path will be triggered.